plugin/client.go

```diff
@@ -25,7 +25,7 @@ func ClientConfig(m discovery.PluginMeta) *plugin.ClientConfig {
 		Managed:          true,
 		Logger:           logger,
 		AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
-		AutoMTLS:         true,
+		AutoMTLS:         false,
```

Comment:
I'm not sure this code (entire file) ever executes from the SDK's perspective, it was removed in the version2 branch.

Reply:
I would hope it doesn't, because it's the client gRPC part - i.e. the part core uses to talk to plugins.

Reply:
Thanks both, you're quite right. This change isn't needed in the SDK and I've raised a PR against Core for the necessary flag: https://github.com/hashicorp/terraform/pull/23563